webui/react/src/pages/ExperimentDetails/ExperimentTrials.settings.ts

```diff
@@ -56,70 +58,104 @@ export interface Settings extends InteractiveTableSettings {
   tableOffset: number;
 }
 
-const config: SettingsConfig = {
+const config: SettingsConfig<Settings> = {
   applicableRoutespace: '/trials',
-  settings: [
-    {
+  settings: {
+    columns: {
       defaultValue: DEFAULT_COLUMNS,
-      key: 'columns',
       skipUrlEncoding: true,
       storageKey: 'columns',
-      type: {
-        baseType: BaseType.String,
-        isArray: true,
-      },
+      type: array(
+        union([
+          literal('action'),
+          literal('id'),
+          literal('state'),
+          literal('totalBatchesProcessed'),
+          literal('bestValidationMetric'),
+          literal('latestValidationMetric'),
+          literal('startTime'),
+          literal('duration'),
+          literal('autoRestarts'),
+          literal('checkpoint'),
+        ]),
+      ),
     },
-    {
+    columnWidths: {
       defaultValue: DEFAULT_COLUMNS.map((col: TrialColumnName) => DEFAULT_COLUMN_WIDTHS[col]),
-      key: 'columnWidths',
       skipUrlEncoding: true,
       storageKey: 'columnWidths',
-      type: {
-        baseType: BaseType.Float,
-        isArray: true,
-      },
+      type: array(number),
     },
-    {
+    compare: {
       defaultValue: false,
-      key: 'compare',
-      type: { baseType: BaseType.Boolean },
+      storageKey: 'compare',
+      type: boolean,
     },
-    {
-      key: 'row',
-      type: { baseType: BaseType.Integer, isArray: true },
+    row: {
+      defaultValue: undefined,
+      storageKey: 'row',
+      type: union([undefinedType, array(number)]),
     },
-    {
+    sortDesc: {
       defaultValue: true,
-      key: 'sortDesc',
       storageKey: 'sortDesc',
-      type: { baseType: BaseType.Boolean },
+      type: boolean,
     },
-    {
+    sortKey: {
       defaultValue: V1GetExperimentTrialsRequestSortBy.ID,
-      key: 'sortKey',
       storageKey: 'sortKey',
-      type: { baseType: BaseType.String },
+      type: union([
+        literal(V1GetExperimentTrialsRequestSortBy.BATCHESPROCESSED),
+        literal(V1GetExperimentTrialsRequestSortBy.BESTVALIDATIONMETRIC),
+        literal(V1GetExperimentTrialsRequestSortBy.DURATION),
+        literal(V1GetExperimentTrialsRequestSortBy.ENDTIME),
+        literal(V1GetExperimentTrialsRequestSortBy.ID),
+        literal(V1GetExperimentTrialsRequestSortBy.LATESTVALIDATIONMETRIC),
+        literal(V1GetExperimentTrialsRequestSortBy.RESTARTS),
+        literal(V1GetExperimentTrialsRequestSortBy.STARTTIME),
+        literal(V1GetExperimentTrialsRequestSortBy.STATE),
+        literal(V1GetExperimentTrialsRequestSortBy.UNSPECIFIED),
+      ]),
     },
-    {
-      key: 'state',
+    state: {
+      defaultValue: undefined,
       storageKey: 'state',
-      type: {
-        baseType: BaseType.String,
-        isArray: true,
-      },
+      type: union([
+        undefinedType,
```

Comment:
how come we need `undefined`? I think you respected the previous implementation, but if we dont need it we can removed it. we can represent undefined as empty array unless we use undefined on purpose